src/nnet3/nnet-example-utils.cc

```diff
@@ -219,5 +219,200 @@ void GetComputationRequest(const Nnet &nnet,
+                (i == num_minibatches - 1 ? extra_right_frames : 0));
+      // the last frame being copied should be the last row of features
+      if (i == num_minibatches - 1 && n == num_chunks - 1)
+        KALDI_ASSERT(src_end == eg.io[feat].features.NumRows());
+      //copy indexes
+      std::copy(eg.io[feat].indexes.begin() + src_begin,
+                eg.io[feat].indexes.begin() + src_end,
+                new_eg.io[feat].indexes.begin() + dst_begin);
+      // modify indexes "t"
+      int32 t = -left_context - (i == 0 ? extra_left_frames : 0);
+      for (int32 j = 0; j < src_end - src_begin; j++, t++)
+        new_eg.io[feat].indexes[dst_begin + j].t = t + output_t_begin[n];
+      // copy corresponding features
+      FilterGeneralMatrixRowsRange(eg.io[feat].features, src_begin,
+                                   src_end, &output_matrices[n]);
+      output_list[n] = &output_matrices[n];
+    }
+    AppendGeneralMatrixRows(output_list, &(new_eg.io[feat].features));
+  }
+}
+
+// Split ivector in eg and store them in *split_egs. It is called by
+// SplitChunk(). It assumes that the space of *split_egs has been allocated by
+// the caller.
+static void SplitIvector(int32 new_chunk_size, int32 num_chunks, int32 feat,
+                         const NnetExample &eg,
+                         std::vector<NnetExample> *split_egs) {
+  KALDI_ASSERT(NumFramesPerChunk(eg.io[feat]) == 1);
+  const int32 num_minibatches = split_egs->size();
+  for (int32 i = 0; i < num_minibatches; i++) {
+    std::vector<GeneralMatrix const*>  output_list(num_chunks);
+    std::vector<GeneralMatrix> output_matrices(num_chunks);
+
+    NnetExample &new_eg = (*split_egs)[i];
+    new_eg.io[feat].name = eg.io[feat].name;
+    new_eg.io[feat].indexes.resize(num_chunks);
+    for (int32 n = 0; n < num_chunks; n++) {
+      int32 src_begin = n, src_end = src_begin + 1, dst_begin = n;
+      // the last frame being copied should be the last row of features
+      if (i == num_minibatches - 1 && n == num_chunks - 1)
+        KALDI_ASSERT(src_end == eg.io[feat].features.NumRows());
+       // copy indexes
+       std::copy(eg.io[feat].indexes.begin() + src_begin,
+                 eg.io[feat].indexes.begin() + src_end,
+                 new_eg.io[feat].indexes.begin() + dst_begin);
+      // copy corresponding features
+      FilterGeneralMatrixRowsRange(eg.io[feat].features, src_begin,
+                                   src_end, &output_matrices[n]);
+      output_list[n] = &output_matrices[n];
+    }
+    AppendGeneralMatrixRows(output_list, &(new_eg.io[feat].features));
+  }
+}
+
+void SplitChunk(int32 new_chunk_size, int32 left_context, int32 right_context,
+                const NnetExample &eg, std::vector<NnetExample> *split_egs) {
+  KALDI_ASSERT(new_chunk_size > 0);
+  int32 old_chunk_size = 0, num_chunks = 0, num_minibatches = 0;
+  std::vector<int32> output_t_begin;
+
+  for (int32 f = 0; f < eg.io.size(); f++) {
+    if (eg.io[f].name == "output") {
```

Comment:
You should not be trying to work out from the name whether it is an output or an input; and names should not be hardcoded at this level of function regardless.  Instead, this function should take the nnet as an argument, and work out from the nnet itself whether the name corresponds to an input or an outputl